core/src/main/java/hudson/model/Slave.java

```diff
@@ -373,7 +375,7 @@ private URLConnection connect() throws IOException {
             return res.openConnection();
         }
 
-        public URL getURL() throws MalformedURLException {
+        public URL getURL() throws IOException {
```

Comment:
> I'd bet that Slave#getURL is popular in plugins

This is `Slave.JnlpJar.getURL`. I cannot imagine many reasons for plugin code to be calling it. I have found seven calls to `readFully`, which calls `getURL` indirectly; and two direct calls, from methods which already threw `IOException`.

> It is binary compatible iff the WAR is exploded correctly.

No, it is binary compatible _always_. An `IOException` thrown from a method not declared in source code to throw it is not considered a lack of binary compatibility. It will just be thrown up the call stack until someone handles it, just like a `RuntimeException`.